lib/reek/code_comment.rb

```diff
@@ -25,14 +27,20 @@ class CodeComment
 
     #
     # @param comment [String] - the original comment as found in the source code
-    #   E.g.:
-    #   "\n        # :reek:Duplication: { enabled: false }\n      "
+    # @param line [Integer] - start of the expression the comment belongs to
+    # @param source [String] - Path to source file or "string"
     #
-    def initialize(comment)
+    def initialize(comment:, line:, source:)
       @original_comment  = comment
       @config            = Hash.new { |hash, key| hash[key] = {} }
 
       @original_comment.scan(CONFIGURATION_REGEX) do |detector, _option_string, options|
+        unless Smells::SmellDetector.valid_detector?(detector)
+          raise BadDetectorInCommentError, detector: detector,
+                                           source: source,
+                                           line: line,
+                                           original_comment: @original_comment
+        end
```

Comment:
I think @waldyr's check is further up in `SmellDetector`, but it would use the same mechanism.